tests/test_config_parsing/test_res_config.py

```diff
@@ -742,3 +831,238 @@ def test_that_failing_to_load_ert_script_with_errors_fails_gracefully(load_state
+        fh.write(test_config_contents)
+
+    with pytest.raises(ConfigValidationError, match="at least 1 arguments"):
+        _ = ErtConfig.from_file(test_config_file_name)
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_include_statements_work():
+    test_config_file_name = "test.ert"
+    test_config_contents = dedent(
+        """
+        NUM_REALIZATIONS  1
+        INCLUDE include.ert
+        """
+    )
+    test_include_file_name = "include.ert"
+    test_include_contents = dedent(
+        """
+        JOBNAME included
+        """
+    )
+    with open(test_config_file_name, "w", encoding="utf-8") as fh:
+        fh.write(test_config_contents)
+    with open(test_include_file_name, "w", encoding="utf-8") as fh:
+        fh.write(test_include_contents)
+
+    ert_config = ErtConfig.from_file(test_config_file_name)
+    assert ert_config.model_config.jobname_format_string == "included"
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_giving_incorrect_queue_name_in_queue_option_fails():
+    test_config_file_name = "test.ert"
+    test_config_contents = dedent(
+        """
+        NUM_REALIZATIONS  1
+        QUEUE_OPTION VOCAL MAX_RUNNING 50
+        """
+    )
+    with open(test_config_file_name, "w", encoding="utf-8") as fh:
+        fh.write(test_config_contents)
+
+    with pytest.raises(ConfigValidationError, match="VOCAL"):
+        _ = ErtConfig.from_file(test_config_file_name)
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_giving_no_keywords_fails_gracefully():
+    test_config_file_name = "test.ert"
+    with open(test_config_file_name, "w", encoding="utf-8") as fh:
+        fh.write("")
+
+    with pytest.raises(ConfigValidationError, match="must be set"):
+        _ = ErtConfig.from_file(test_config_file_name)
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_recursive_defines_fails_gracefully_and_logs(caplog):
+    test_config_file_name = "test.ert"
+    test_config_contents = dedent(
+        """
+        DEFINE <A> <A>
+        NUM_REALIZATIONS  <A>
+        """
+    )
+    with open(test_config_file_name, "w", encoding="utf-8") as fh:
+        fh.write(test_config_contents)
+
+    with pytest.raises(ConfigValidationError, match="integer"), caplog.at_level(
+        logging.WARNING
+    ):
+        _ = ErtConfig.from_file(test_config_file_name)
+        assert len(caplog.records) == 1
+        assert "max iterations" in caplog.records[0]
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_define_string_quotes_are_removed():
+    test_config_file_name = "test.ert"
+    test_config_contents = dedent(
+        """
+        DEFINE <A> "A"
+        NUM_REALIZATIONS 1
+        """
+    )
+    with open(test_config_file_name, "w", encoding="utf-8") as fh:
+        fh.write(test_config_contents)
+
+    ert_Config = ErtConfig.from_file(test_config_file_name)
+    assert ert_Config.substitution_list.get("<A>") == "A"
```

Comment:
should this (it looks like new behaviour?!) be documented? do we in general document the changes in the behaviour of DEFINE or other things? maybe we don't need to publish them, but could gather them in a document already, while we remember

Reply:
This is not a change, just an added test